ledger/internal/eval_test.go

```diff
@@ -214,45 +217,51 @@ func TestCowCompactCert(t *testing.T) {
 		&ml, bookkeeping.BlockHeader{}, config.Consensus[protocol.ConsensusCurrentVersion],
 		0, ledgercore.AccountTotals{}, 0)
 
-	certType = protocol.CompactCertType(1234) // bad cert type
-	err := c0.compactCert(certRnd, certType, cert, atRound, validate)
+	spType = protocol.StateProofType(1234) // bad stateproof type
+	stateProofTx := transactions.StateProofTxnFields{
+		StateProofIntervalLastRound: atRound,
+		StateProofType:              spType,
+		StateProof:                  stateProof,
+		Message:                     msg,
+	}
+	err := apply.StateProof(stateProofTx, atRound, c0, validate)
```

Comment:
I wish the error types were asserted here (and below). It could still be in one line, with `require.ErrorIs`

Reply:
it is a little bit tricky to use `errorIs` since the errors are defined in ledger/internal (even if we export the errors it would be a problem since it is internal package). how would suggest we do it?
we can copy and the content of the errors

Reply:
But isn't this test in `ledger/internal` too?

If I'm misunderstanding the problem, how about just test for some string in the error for now `require.Contains(t, err.Error(), "some bit of the error")` but put a note in to use errors.Is once we move things things out of `internal` I believe that is planned.

Finally, I don't mind that string check as a permanent solution if the errors don't have names. I thought I saw your code exporting a lot of error variables, so I thought it would be easy to do `error.Is`. But if not, string check is fine. I've just seen too many tests that pass because _something_ fails, but not the thing that was supposed to be tested.

Reply:
got it
we would fix that in this PR
https://github.com/algorand/go-algorand/pull/4355